CMake/resolve_dependency_modules/folly.cmake

```diff
@@ -48,8 +48,13 @@ FetchContent_Declare(
   folly
   URL ${FOLLY_SOURCE_URL}
   URL_HASH SHA256=${VELOX_FOLLY_BUILD_SHA256_CHECKSUM})
-# Fetch the content using previously declared details
+
+if(ON_APPLE_M1)
```

Comment:
Nice - I think this solves #3453

Reply:
Yes it does. I will open an issue with folly too to fix the root cause.